src/wrappers/themis/python/pythemis/smessage.py

```diff
@@ -24,82 +25,115 @@
+                              "Secure Message: missing public key")
+        if not _private_key(private_key):
+            raise ThemisError(THEMIS_CODES.FAIL,
+                              "Secure Message: invalid private key")
+        if not _public_key(peer_public_key):
+            raise ThemisError(THEMIS_CODES.FAIL,
+                              "Secure Message: invalid public key")
         self.private_key = private_key
         self.peer_public_key = peer_public_key
 
     def wrap(self, message):
         encrypted_message_length = c_int(0)
-        res = themis.themis_secure_message_wrap(
+        res = themis.themis_secure_message_encrypt(
             self.private_key, len(self.private_key),
             self.peer_public_key, len(self.peer_public_key),
             message, len(message), None, byref(encrypted_message_length))
         if res != THEMIS_CODES.BUFFER_TOO_SMALL:
-            raise ThemisError(res, "Secure Message failed encrypting")
+            raise ThemisError(res, "Secure Message failed to encrypt")
         encrypted_message = create_string_buffer(encrypted_message_length.value)
-        res = themis.themis_secure_message_wrap(
+        res = themis.themis_secure_message_encrypt(
             self.private_key, len(self.private_key),
             self.peer_public_key, len(self.peer_public_key),
             message, len(message), encrypted_message,
             byref(encrypted_message_length))
         if res != THEMIS_CODES.SUCCESS:
-            raise ThemisError(res, "Secure Message failed encrypting")
+            raise ThemisError(res, "Secure Message failed to encrypt")
 
         return string_at(encrypted_message, encrypted_message_length.value)
 
     def unwrap(self, message):
         plain_message_length = c_int(0)
-        res = themis.themis_secure_message_unwrap(
+        res = themis.themis_secure_message_decrypt(
             self.private_key, len(self.private_key),
             self.peer_public_key, len(self.peer_public_key),
             message, len(message), None, byref(plain_message_length))
         if res != THEMIS_CODES.BUFFER_TOO_SMALL:
-            raise ThemisError(res, "Secure Message failed decrypting")
+            raise ThemisError(res, "Secure Message failed to decrypt")
         plain_message = create_string_buffer(plain_message_length.value)
-        res = themis.themis_secure_message_unwrap(
+        res = themis.themis_secure_message_decrypt(
             self.private_key, len(self.private_key),
             self.peer_public_key, len(self.peer_public_key),
             message, len(message), plain_message, byref(plain_message_length))
         if res != THEMIS_CODES.SUCCESS:
-            raise ThemisError(res, "Secure Message failed decrypting")
+            raise ThemisError(res, "Secure Message failed to decrypt")
         return string_at(plain_message, plain_message_length.value)
 
 
 def ssign(private_key, message):
     encrypted_message_length = c_int(0)
```

Comment:
can you add `private_key` length check and error message here as well? I think it would be nice addition to this PR

Reply:
Yeah, it makes sense to check keys for sign/verify mode as well. I'll add the checks.